server/isomorphic-routing/index.js

```diff
@@ -93,15 +93,10 @@ function compose( ...functions ) {
 	return functions.reduceRight( ( composed, f ) => () => f( composed ), () => {} );
 }
 
-export function getCacheKey( context ) {
-	if ( isEmpty( context.query ) || isEmpty( context.cacheQueryKeys ) ) {
-		return context.pathname;
+export function getNormalizedPath( pathname, query ) {
+	if ( isEmpty( query ) ) {
+		return pathname;
 	}
 
-	const cachedQueryParams = pick( context.query, context.cacheQueryKeys );
-	return (
-		context.pathname +
-		'?' +
-		stringify( cachedQueryParams, { sort: ( a, b ) => a.localCompare( b ) } )
-	);
+	return pathname + '?' + stringify( query, { sort: ( a, b ) => a.localeCompare( b ) } );
```

Comment:
pathname will need to be escaped, I think we'll want `encodeURIComponent`. See failing tests.

Reply:
🤔 Not so simple. Maybe I'm being overzealous with my testing here and we should expect to receive paths which include escaped characters for anything with special significance (`?`)